### PR TITLE
[5.2] Add getAuthIdentifierName to Authenticable trait

### PR DESCRIPTION
**Pull request by patrickbrouwers**
Replaces https://github.com/laravel/framework/pull/10291

### CODE REVIEW

File: src/Illuminate/Auth/Authenticatable.php [hunk 1]
```diff
@@ -5,6 +5,16 @@
 trait Authenticatable
 {
     /**
+     * Get the name of the unique identifier for the user.
+     *
+     * @return string
+     */
+    public function getAuthIdentifierName()
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Authenticatable.php]: Breaking change.

File: src/Illuminate/Contracts/Auth/Authenticatable.php [hunk 2]
```diff
@@ -5,6 +5,13 @@
 interface Authenticatable
 {
     /**
+     * Get the name of the unique identifier for the user.
+     *
+     * @return string
+     */
+    public function getAuthIdentifierName();
```
Comment (GrahamCampbell) [inline src/Illuminate/Contracts/Auth/Authenticatable.php]: Breaking change.

Comment (GrahamCampbell) [inline src/Illuminate/Contracts/Auth/Authenticatable.php]: Oh right, I see, this is targeted at 5.2. This is fine then. ;)

Comment (GrahamCampbell) [inline src/Illuminate/Contracts/Auth/Authenticatable.php]: I should read stuff more carefully, lol.

Comment (patrickbrouwers) [inline src/Illuminate/Contracts/Auth/Authenticatable.php]: Np :)

**Pull request merged**